### PR TITLE
Syncfile

### PR DESCRIPTION
**Pull request by karissa**
Initial pass of implementing the 'Sync to USB' button. This definitely needs a close look -- I am a bit out of my depth with the existing ducks/epics model.

Depends upon merging and publishing this PR: https://github.com/digidem/mapeo-sync/pull/4 in a new version of mapeo-server.

### CODE REVIEW

File: src/components/Views/SyncView/SyncView.js [hunk 1]
```diff
@@ -155,6 +156,10 @@ class SyncView extends React.Component<
 
     const keyExtractor = (item, index) => item.id;
 
+    const handleUSBPress = () => {
+      startSync({filename: '/sdcard/Android/data/com.mapeomobile/static/data.syncfile'})
```
Comment (czengg) [inline src/components/Views/SyncView/SyncView.js]: Is this path consistent for all Android phones?

File: src/epics/devices.js [hunk 2]
```diff
@@ -65,11 +65,12 @@ export const syncStartEpic = (action$: ActionsObservable<any>) =>
   action$
     .ofType(SYNC_START)
     .filter(action => action.status === 'Start')
-    .flatMap(action =>
+    .flatMap(action => {
+      if (action.meta.filename) return Sync.toFile(action.meta.filename)
```
Comment (czengg) [inline src/epics/devices.js]: this should be 

```
const syncing = action.meta.filename ? Sync.toFile(action.meta.filename) : Sync.start(action.meta);
return syncing.map(resp => syncStart(meta, resp);
```

File: src/components/Views/SyncView/SyncView.js (hunk 1 above)
Comment (karissa) [reply src/components/Views/SyncView/SyncView.js]: no :P we definitely want to change this

**Comment by noffle** [comment]
You can use mapeo-sync@3.1.0 now!

File: src/rnnodeapp/index.js [hunk 3]
```diff
@@ -62,7 +62,8 @@ function start() {
   const route = Router(db, media, {
     host: HOST,
     media: { mode: 'push' },
-    staticRoot: STATIC_PATH
+    staticRoot: STATIC_PATH,
+    writeFormat: 'osm-p2p-syncfile'
```
Comment (noffle) [inline src/rnnodeapp/index.js]: could we do `sync: { writeFormat: 'osm-p2p-syncfile' }` instead? same on your mapeo-desktop PR -- same way we're namespacing the `media` opts

Comment (karissa) [reply src/rnnodeapp/index.js]: sure sounds good!

**Comment by noffle** [comment]
nevermind, I see that we're passing in opts directly to mapeo-sync

**Comment by karissa** [comment]
ok cool @noffle .

File: src/api/sync.js [hunk 4]
```diff
@@ -20,6 +20,12 @@ class Sync {
       method: 'GET',
       route: `/sync/start?host=${device.host}&port=${device.port}`
     });
+
+  static dataset = (dataset: string) =>
+    blankRequest({
+      method: 'GET',
+      route: `/sync/start?dataset=${dataset}`
```
Comment (karissa) [inline src/api/sync.js]: this should add the `filepath` for the USB location

**Pull request closed without merging**